metrics/prometheus/prometheus.go

```diff
@@ -111,6 +111,29 @@ func (g prometheusGauge) Add(delta float64) {
 	g.GaugeVec.With(prometheus.Labels(g.Pairs)).Add(delta)
 }
 
+// RegisterCallbackGauge registers a Gauge with Prometheus whose value is
+// determined at collect time by the passed callback function. The callback
+// determines the value, and fields are ignored, so RegisterCallbackGauge
+// returns nothing.
+func RegisterCallbackGauge(namespace, subsystem, name, help string, callback func() float64) {
+	RegisterCallbackGaugeWithLabels(namespace, subsystem, name, help, prometheus.Labels{}, callback)
+}
+
+// RegisterCallbackGaugeWithLabels is the same as RegisterCallbackGauge, but
+// attaches a set of const label pairs to the metric.
+func RegisterCallbackGaugeWithLabels(namespace, subsystem, name, help string, constLabels prometheus.Labels, callback func() float64) {
```

Comment:
Actually, in this specific example, a `GaugeFunc` is a gauge without any variable labels, totally one-dimensional. For doing callback-based multidimensional metrics with label values determined at callback time, implement the `prometheus.Collector` interface.